sphinx_click/ext.py

```diff
@@ -104,18 +104,18 @@ def _write_opts(opts: ty.List[str]) -> str:
     else:
         show_default = ctx.show_default
 
-    if isinstance(show_default, str):
+    if show_default and isinstance(show_default, str):
         # Starting from Click 7.0 show_default can be a string. This is
         # mostly useful when the default is not a constant and
         # documentation thus needs a manually written string.
         extras.append(':default: ``%s``' % show_default)
```

Comment:
Can't we just use `%r` instead of `%s` below? As things stand, this will ignore a default of `''` (the empty string).

```suggestion
    if isinstance(show_default, str):
        # Starting from Click 7.0 show_default can be a string. This is
        # mostly useful when the default is not a constant and
        # documentation thus needs a manually written string.
        extras.append(':default: ``%r``' % show_default)
```